### PR TITLE
Update of jupyter notebooks for model evaluation

### PR DESCRIPTION
**Pull request by ehellbar**
Adding and updating notebooks and helper file for model performance and kernel visualization.

### CODE REVIEW

**Comment by ehellbar** [comment]
These are just some basic examples for plots which can be interesting. Of course, depending on the study or the use case, they need to be modified. I would suggest to create notebooks of similar style for every major study that we perform, e.g. like it was done for the GPU comparison or for the studies with large number of events and different grid sizes as presented by Maja over the last weeks.

File: tpcwithdnn/kernel_visualization.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+#pylint: disable=too-many-arguments, too-many-function-args
+import numpy as np
+import pandas as pd
+from tensorflow.keras.models import model_from_json
+from tpcwithdnn.symmetry_padding_3d import SymmetryPadding3d
+from tpcwithdnn.data_loader import load_data_original
+
+
+def get_full_charge_response(n_phi, n_r, n_z, model_files, input_dir, \
```
Comment (saganatt) [inline tpcwithdnn/kernel_visualization.py:9]: \ is not needed here, same for other functions

File: tpcwithdnn/kernel_visualization.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    dist_names = np.array(['R', 'Rphi', 'Z'])[np.array([output_pred[0], \
+                                                        output_pred[1], \
+                                                        output_pred[2]]) > 0]
+    column_names = np.array(['phi', 'r', 'z',
+                             'flucSC', 'meanSC'])
+    for dist_name in dist_names:
+        column_names = np.append(column_names, ['flucDist' + dist_name + 'Pred'])
+
+    # load position arrays
+    arr_r_pos, arr_phi_pos, arr_z_pos, \
```
Comment (saganatt) [inline tpcwithdnn/kernel_visualization.py:29]: you can write just: `arr_r_pos, arr_phi_pos, arr_z_pos, *_ = load_data_original(input_dir, [0,0])`

File: tpcwithdnn/kernel_visualization.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    _, _, \
+    _, _ = \
+        load_data_original(input_dir, [0,0])
+    arr_r_pos = arr_r_pos[arr_z_pos>0]
+    arr_phi_pos = arr_phi_pos[arr_z_pos>0]
+    arr_z_pos = arr_z_pos[arr_z_pos>0]
+
+    # define input densities, create data frame and fill with prediction
+    input_single = np.zeros((1, n_phi, n_r, n_z, 2))
+    input_single[0, i_phi, i_r, :, 0] = input_sc_mean
```
Comment (saganatt) [inline tpcwithdnn/kernel_visualization.py:95]: Most of the code is repeated, these two lines seem to be the only difference between the 3 functions.
What about defining a single function for data processing + 3 wrappers?

**Review by saganatt — CHANGES_REQUESTED** [review]
Hi Ernst, thank you very much for the update.

I will look more and try to understand the physics interpretation of these plots.

In the meantime, could you check my comments to the code?

**Pull request closed without merging**